frontend/packages/dev-console/src/components/AddPage.tsx

```diff
@@ -34,29 +34,25 @@ const handleProjectCreate = (project: K8sResourceKind) =>
 
 const EmptyStateLoader: React.FC<EmptyStateLoaderProps> = ({ resources, loaded, loadError }) => {
   const [noWorkloads, setNoWorkloads] = React.useState(false);
-  const knativeData = _.get(resources, ['knativeService', 'data'], null);
+  const daemonSets = resources.daemonSets?.data;
```

Comment:
According to `EmptyStateLoaderProps` `resources` are optional as well. So, there might be a case where the resources object is `undefined`. Safer to do -
```suggestion
  const daemonSets = resources?.daemonSets?.data;
```

Reply:
As I understood that, resources is marked only as optional because the the `EmptyStateLoader` was wrapped in the `Firehose` component. This component forces that `resources` is always defined, but could not be marked as required because the EmptyStateLoader JSX code would fail. But I can add these ? anyway, for me this is also okay.

https://github.com/openshift/console/blob/master/frontend/packages/dev-console/src/components/AddPage.tsx#L117-L119

Reply:
Actually we have `resources` optional here because sometimes `Firehose` doesn't inject the props on time which then breaks the UI. I've seen similar issue in other components as well - https://github.com/openshift/console/pull/5365#discussion_r423964197. That's why it's better to add an extra check.

Reply:
Thanks @rohitkrai03 for that hint. :+1:

I added the optional operator `?` now also to resources.